src/mpid/ch4/netmod/ofi/ofi_dynproc.c

```diff
@@ -6,9 +6,11 @@
 #include "mpidimpl.h"
 #include "ofi_impl.h"
 
+#define DISCONNECT_CONTEXT_ID 0xF000
```

Comment:
I see this value was used previously, so I think it is OK. A comment might be good to add to say why this is safe to use. That is, why it won't collide with other context ids. Or if it might collide, a FIXME comment to go back and add a proper fix.

Reply:
> Could you explain more what is happening in this commit, and if this is a new concern or is from preexisting code? This looks mostly like refactoring, in which case it should be OK.

Previously for children connections, it does a *blocking* send -- issues the send then progress loop until it completes. For parent connection, it does *non-blocking* recv -- issues the recv, add the request to an array. And the progress loop waiting for the request happens separately, i.e. after all the children send completes.

The new code does both *blocking* send (no change) and *blocking* recv (change). Thus raising the question of whether we may have a potential deadlock. So it is a new concern.

I believe all the processes form a strict tree-like structure, i.e. no process can be both ancestor/descendent of the same process forming `send-recv` cycles. Thus all the blocking send and recv should be able to complete. Thus the concern can be dismissed.

The code is much simpler to maintain with blocking send/recv as they can be easily moved around. In the future, we probably want to allow processes to disconnect before `MPI_Finalize`, so the simpler function wrap paves the way.